### ISSUE TITLE
Repeated Crashing After Home Assistant Restart (while internet wasn't yet connected ?)

### ISSUE THREAD

**Issue by Jimmypl007** [brenner-tobias/addon-cloudflared#162]
### The problem
Since having to shutdown pi and boot back up the addon no longer will launch without crashing (over and over).

_im no genius but I'm guessing the errors relating to the crash in the log are these_
**2022-08-19T09:00:20Z INF Starting metrics server on [::]:36500/metrics
2022-08-19T09:00:20Z INF You requested 4 HA connections but I can give you at most 0.
2022-08-19T09:00:20Z INF Tunnel server stopped
2022-08-19T09:00:20Z ERR Initiating shutdown error="there are no free edge addresses left to resolve to"
2022-08-19T09:00:21Z INF Metrics server stopped
there are no free edge addresses left to resolve to**

I had to turn the power off to the house for some work, I properly shut down home Assistant and unplugged the pi.
when I turned the pi back on the Router hadn't yet connected to the internet (it had also been powered off).
Before this, the addon was working perfectly, after that it's a no-go.

As you can see in the log below in an attempt to 'dodge' the problem, I re-setup the tunnel.
I had previously reset all Cloudflare files as the documentation shows.
And after a successful tunnel setup, it crashed shortly after.
With "watchdog" enabled obviously when the addon crashes it restarts it, and the addon crashes at the same place (minus the tunnel setup) and repeats.
this log is without watchdog enabled.

### What version of Cloudflared has the issue?

2.0.12

### What type of installation are you running?

Home Assistant OS

### Anything in the logs that might be useful for us?

```txt
cont-init: info: running /etc/cont-init.d/00-banner.sh
-----------------------------------------------------------
 Add-on: Cloudflared
 Use a Cloudflared tunnel (formerly Argo Tunnel) to remotely connect to Home Assistant without opening any ports
-----------------------------------------------------------
 Add-on version: 2.0.12
 You are running the latest version of this add-on.
 System: Home Assistant OS 8.4  (aarch64 / raspberrypi4-64)
 Home Assistant Core: 2022.8.6
 Home Assistant Supervisor: 2022.08.3
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
cont-init: info: /etc/cont-init.d/00-banner.sh exited 0
cont-init: info: running /etc/cont-init.d/01-log-level.sh
cont-init: info: /etc/cont-init.d/01-log-level.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-cloudflared-log: starting
s6-rc: info: service init-cloudflared-log successfully started
s6-rc: info: service init-cloudflared-config: starting
[16:59:11] INFO: Checking Add-on config...
[16:59:12] INFO: Checking for existing certificate...
[16:59:12] NOTICE: No certificate found
[16:59:12] INFO: Creating new certificate...
[16:59:12] NOTICE: 
[16:59:12] NOTICE: Please follow the Cloudflare Auth-Steps:
[16:59:12] NOTICE: 
Please open the following URL and log in with your Cloudflare account:
https://dash.cloudflare.com/argotunnel?callback=https_removed_
Leave cloudflared running to download the cert automatically.
2022-08-19T09:00:11Z INF Waiting for login...
You have successfully logged in.
If you wish to copy your credentials to a server, they have been saved to:
/root/.cloudflared/cert.pem
[17:00:11] INFO: Authentication successfull, moving auth file to the '/data' folder
[17:00:11] INFO: Checking for existing certificate...
[17:00:11] INFO: Existing certificate found
[17:00:11] INFO: Checking for existing tunnel...
[17:00:11] NOTICE: No tunnel file found
[17:00:11] INFO: Creating new tunnel...
Tunnel credentials written to /data/tunnel.json. Keep this file secret. To revoke these credentials, delete the tunnel.
Created tunnel homeassistant-3 with id _removed_
[17:00:14] INFO: Checking for existing tunnel...
[17:00:14] INFO: Existing tunnel with ID _removed_ found
[17:00:14] INFO: Checking if existing tunnel matches name given in config
[17:00:15] INFO: Existing Cloudflare tunnnel name matches config, proceeding with existing tunnel file
[17:00:15] INFO: Creating config file...
[17:00:17] INFO: Validating config file...
Validating rules from /tmp/config.json
OK
[17:00:18] INFO: Creating new DNS entry _removed_.casa...
2022-08-19T09:00:19Z INF Added CNAME _removed_.casa which will route to this tunnel tunnelID=_removed_
[17:00:19] INFO: Finished setting-up the Cloudflare tunnel
s6-rc: info: service init-cloudflared-config successfully started
s6-rc: info: service cloudflared: starting
s6-rc: info: service cloudflared successfully started
s6-rc: info: service healthcheck: starting
s6-rc: info: service healthcheck successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[17:00:19] INFO: Starting Cloudflared Healthcheck for Home-Assistant add-on.
[17:00:20] INFO: Connecting Cloudflared Tunnel...
2022-08-19T09:00:20Z INF Starting tunnel tunnelID=_removed_
2022-08-19T09:00:20Z INF Version 2022.8.2
2022-08-19T09:00:20Z INF GOOS: linux, GOVersion: go1.18.5, GoArch: arm64
2022-08-19T09:00:20Z INF Settings: map[config:/tmp/config.json cred-file:/data/tunnel.json credentials-file:/data/tunnel.json loglevel:info metrics:0.0.0.0:36500 no-autoupdate:true origincert:/data/cert.pem]
2022-08-19T09:00:20Z INF Generated Connector ID: _removed_
2022-08-19T09:00:20Z ERR Unable to lookup protocol. Defaulting to `http2`. If this fails, you can set `--protocol h2mux` in your cloudflared command. error="lookup protocol-v2.argotunnel.com on 127.0.0.11:53: server misbehaving"
2022-08-19T09:00:20Z INF Initial protocol http2
2022-08-19T09:00:20Z INF Starting metrics server on [::]:36500/metrics
2022-08-19T09:00:20Z INF You requested 4 HA connections but I can give you at most 0.
2022-08-19T09:00:20Z INF Tunnel server stopped
2022-08-19T09:00:20Z ERR Initiating shutdown error="there are no free edge addresses left to resolve to"
2022-08-19T09:00:21Z INF Metrics server stopped
there are no free edge addresses left to resolve to
[17:00:21] WARNING: cloudflared crashed, halting add-on
s6-rc: info: service legacy-services: stopping
[17:00:21] INFO: cloudflared stoped, restarting...
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service healthcheck: stopping
[17:00:21] INFO: healthcheck stoped, restarting...
s6-rc: info: service healthcheck successfully stopped
s6-rc: info: service cloudflared: stopping
[17:00:21] INFO: cloudflared stoped, restarting...
s6-rc: info: service cloudflared successfully stopped
s6-rc: info: service init-cloudflared-config: stopping
s6-rc: info: service init-cloudflared-config successfully stopped
s6-rc: info: service init-cloudflared-log: stopping
s6-rc: info: service init-cloudflared-log successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

### Steps to reproduce the issue

attempting to start the addon without the internet available _seems_ to have got me here.

### Additional information

I am not certain what type of installation I am running of HA it seems the steps to find it has changed. or I am blind..

**Comment by nemo-df** [comment]
I observe the same situation in myself. Addon does not start. Although the argo tunnel has been created

`[04:17:45] INFO: Creating new tunnel...
failed to create tunnel: Create Tunnel API call failed: tunnel with name already exists
[04:17:47] FATAL: Failed to create tunnel.
    Please check the Cloudflare Teams Dashboard for an existing tunnel with the name homeassistant and delete it:
    https://dash.teams.cloudflare.com/ Access / Tunnels
s6-rc: warning: unable to start service init-cloudflared-config: command exited 1
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
prog: fatal: stopping the container.
s6-rc: info: service init-cloudflared-log: stopping
s6-rc: info: service init-cloudflared-log successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
`

**Comment by Jimmypl007** [comment]
> I observe the same situation in myself. Addon does not start. Although the argo tunnel has been created
> 
> `[04:17:45] INFO: Creating new tunnel...
> failed to create tunnel: Create Tunnel API call failed: tunnel with name already exists
> [04:17:47] FATAL: Failed to create tunnel.
>     Please check the Cloudflare Teams Dashboard for an existing tunnel with the name homeassistant and delete it:
>     https://dash.teams.cloudflare.com/ Access / Tunnels
> s6-rc: warning: unable to start service init-cloudflared-config: command exited 1
> /run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
> prog: fatal: stopping the container.
> s6-rc: info: service init-cloudflared-log: stopping
> s6-rc: info: service init-cloudflared-log successfully stopped
> s6-rc: info: service legacy-cont-init: stopping
> s6-rc: info: service legacy-cont-init successfully stopped
> s6-rc: info: service fix-attrs: stopping
> s6-rc: info: service fix-attrs successfully stopped
> s6-rc: info: service s6rc-oneshot-runner: stopping
> s6-rc: info: service s6rc-oneshot-runner successfully stopped`

I thought there was a fix for that occuring but as a temporary fix for that just try going to config and changing the tunnel name (I add a number after it "homeassistant-2") and than re setup the tunnel..

**Comment by Jimmypl007** [comment]
After leaving it overnight and then starting the Cloudflare addon manually, it launched successfully with no crashes and has been running fine now for a few hours.

It might not be related but this will be the first time I manually started the addon since it kept crashing, every other time I rebooted (turned off, unplugged, turned back on) home assistant. OR it was auto restarted with the crash due to watchdog OR i changed a config setting and it restarted when saved.

whatever the case the problem seems to have gone away for now..
